modules/tensor_mechanics/include/materials/ADLAROMANCEStressUpdateBase.h

```diff
@@ -11,40 +11,70 @@
 
 #include "ADRadialReturnCreepStressUpdateBase.h"
 
-#include "LAROMData.h"
+#define usingADLAROMANCEStressUpdateBase                                                           \
+  usingRadialReturnCreepStressUpdateBaseMembers;                                                   \
+  using ADLAROMANCEStressUpdateBase<compute_stage>::getTransform;                                  \
+  using ADLAROMANCEStressUpdateBase<compute_stage>::getTransformCoefs;                             \
+  using ADLAROMANCEStressUpdateBase<compute_stage>::getInputLimits;                                \
+  using ADLAROMANCEStressUpdateBase<compute_stage>::getCoefs
 
-template <ComputeStage compute_stage>
-class ADLAROMCreepStressUpdate;
+template <ComputeStage>
+class ADLAROMANCEStressUpdateBase;
 
-declareADValidParams(ADLAROMCreepStressUpdate);
+declareADValidParams(ADLAROMANCEStressUpdateBase);
 
-/**
- * Class to call the Reduced Order Model to predict the behavior of creep
- */
+enum class ROMInputTransform
+{
+  LINEAR,
+  LOG,
+  EXP
+};
 
 template <ComputeStage compute_stage>
-class ADLAROMCreepStressUpdate : public ADRadialReturnCreepStressUpdateBase<compute_stage>
+class ADLAROMANCEStressUpdateBase : public ADRadialReturnCreepStressUpdateBase<compute_stage>
 {
 public:
-  ADLAROMCreepStressUpdate(const InputParameters & parameters);
+  ADLAROMANCEStressUpdateBase(const InputParameters & parameters);
 
 protected:
+  virtual void initialSetup() override;
+
   virtual void initQpStatefulProperties() override;
+
   virtual ADReal computeResidual(const ADReal & effective_trial_stress,
                                  const ADReal & scalar) override;
+
   virtual ADReal computeDerivative(const ADReal & /*effective_trial_stress*/,
                                    const ADReal & /*scalar*/) override
   {
     return _derivative;
   }
+
   virtual void computeStressFinalize(const ADRankTwoTensor & plastic_strain_increment) override;
 
   /**
    * Compute the limiting value of the time step for this material
+   *
    * @return Limiting time step
    */
   virtual Real computeTimeStepLimit() override;
 
+  /**
+   *Computes the ROM Strain rate
+   @param dt Timestep size
+   @param mobile_dislocations_old Mobile dislocation density from previous timestep
+   @param immobile_dislocations_old Immobile dislocation density from previous timestep
+   @param trial_stress Trial stress from radial return method
+   @param effective_strain_old Effective strain from the previous timestep
+   @param temperature Temperature variable value
+   @param environmental Environmental variable value
+   @param mobile_dislocation_increment Mobile dislocation density incremental change
+   @param immobile_dislocation_increment Immobile dislocation density incremental change
+   @param rom_effective_strain ROM calculated effective strain
+   @param rom_effective_strain_derivative Derivative of ROM calculated effective strain with respect
+   to stress
```

Comment:
Do these parameter definitions need `*`s in front to render properly in doxygen?